dsl/src/test/scala/org/http4s/dsl/PathSpec.scala

```diff
@@ -12,31 +12,27 @@ package dsl
 
 import cats.effect.IO
 import org.http4s.Uri.uri
+import org.http4s.Uri.Path
+import org.http4s.Uri.Path.{Root, Segment}
 import org.http4s.dsl.io._
-import org.scalacheck.Arbitrary
 import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
 
 class PathSpec extends Http4sSpec {
-  implicit val arbitraryPath: Arbitrary[Path] =
-    Arbitrary {
-      arbitrary[List[String]].map(Path(_))
-    }
+  implicit val arbitraryPath: Gen[Path] =
+    arbitrary[List[String]]
+      .map(_.foldLeft(Path.Root)((acc, e) => acc.copy(segments = acc.segments :+ Segment(e))))
 
   "Path" should {
-    "/foo/bar" in {
-      Path("/foo/bar") must_== Path("foo", "bar")
-    }
-
-    "foo/bar" in {
-      Path("foo/bar") must_== Path("foo", "bar")
-    }
-
-    "//foo/bar" in {
-      Path("//foo/bar") must_== Path("", "foo", "bar")
+    "~ extractor on Path without Root" in {
+      (Path(Vector(Segment("foo.json"))) match {
```

Comment:
you can use `path"foo.json"` or `Path.fromString`

Reply:
@hamnis I am so sorry. I thought I had changed all to `Path.fromString`. Apparently I ended up doing the changes only inside the Matrix extractor tests. Will change it and push

Reply:
done